hikvision-nvr/operations.py

```diff
@@ -0,0 +1,176 @@
+"""
+Copyright start
+MIT License
+Copyright (c) 2024 Fortinet Inc
+Copyright end
+"""
+
+import json
+from os.path import join, splitext
+
+import requests
+import xmltodict
+from connectors.core.connector import get_logger, ConnectorError
+from connectors.cyops_utilities.builtins import upload_file_to_cyops
+from django.conf import settings
+
+from .constants import (DEFAULT_EXTENSION, LOG_SEARCH_XML_PAYLOAD)
+
+error_msgs = {
+    400: 'Bad/Invalid Request',
+    401: 'Unauthorized: Invalid credentials or token provided failed to authorize',
+    403: 'Access Denied',
+    404: 'Not Found',
+    500: 'Internal Server Error',
+    503: 'Service Unavailable'
+}
+
+logger = get_logger('hikvision-nvr')
+
+
+class hikvision(object):
+    def __init__(self, config, *args, **kwargs):
+        self.username = config.get('username')
+        self.password = config.get('password')
+        self.url = config.get('server_url', '').strip('/')
+        if not self.url.startswith('https://') and not self.url.startswith('http://'):
+            self.url = 'http://{0}'.format(self.url)
+        elif self.url.startswith('https://'):
+            self.url = self.url.replace('https://', 'http://')
+        else:
+            self.url = self.url
+
+    def make_rest_call(self, endpoint, headers=None, params=None, data=None, method='GET'):
+        try:
+            url = self.url + endpoint
+            logger.debug('Rest API Endpoint {0}'.format(url))
+            auth = requests.auth.HTTPDigestAuth(self.username, self.password)
+            response = requests.request(method, url, auth=auth, data=data)
+            logger.debug('Rest API Status Code: {0}'.format(response.status_code))
+            # logger.debug('Rest API Response: {0}'.format(response.text))
```

Comment:
this comment line can be removed, same log is there.
also add curl utils log

Reply:
added curl_utils